### DISCUSSION TITLE
Requirement of `errno` usage in `ERR_raise` cases

### DISCUSSION THREAD

**Discussion by dotnwat** [openssl/openssl#26062] [Q&A] [answered]
The OpenSSL document contains this statement that "errno must be passed to ERR_raise/ERR_raise_data":

```
ERR_LIB_SYS

This "library code" indicates that a system error is being reported. In this case, the reason code given to ERR_raise() and ERR_raise_data() must be [errno(3)](https://docs.openssl.org/3.1/man3/errno).

ERR_raise(ERR_LIB_SYS, errno);
```

However, I find this to be slightly ambiguous. I can imagine three different interpretations:

1. The actual value of `errno` needs to be passed in following a system call error
2. A valid errno value (e.g. values in `errno.h`) must be passed in
3. ERR_raise/ERR_raise_data expect (1) or (2), but also independently will access the errno thread local variable

My hope is that (2) is the correct interpretation. In most cases yes this will be the current value of `errno`, but at least with C++ we get `std::system_error` and can extract out the system error code, but otherwise `errno` may not be reliably read.

**Comment by t8m** [comment, marked as answer]
> My hope is that (2) is the correct interpretation.

Yes, it is.
